### PR TITLE
Show subsequent step under banner

### LINKED ISSUE #469: Show subsequent step under turn banner
When the next two maneuvers are sufficiently close together, we should display a mini banner beneath the turn banner for the second maneuver. The mini banner would parallel the “linked” voice instructions that the SDK already provides, but we could display it for slightly less closely spaced maneuvers as well.

/cc @mapbox/navigation-ios

### PR DESCRIPTION
**Pull request by frederoni**
Fixes #469 

Adds a banner for the subsequent step under the instructions banner.

@bsudekum @ericrwolfe @JThramer

### CODE REVIEW

**Comment by bsudekum** [comment]
I'm not sold on the design here yet. I think the main issue is that we're not telling the user what this new view is. I really like Google's design because it's simple, informative and it's not close to visually equal with the main banner:

image

At the point where we're showing this view to the user, it's not actually necessary to tell them the name of the road. One might argue actually giving the user two road names (perhaps even 3 with a destination) is too much and confusing. My vote is still for `Then: maneuver-arrow`.

File: MapboxNavigation/RouteMapViewController.swift [hunk 1]
```diff
@@ -388,6 +388,35 @@ class RouteMapViewController: UIViewController {
         instructionsBannerView.maneuverView.step = routeProgress.currentLegProgress.upComingStep
     }
     
+    func updateNextBanner(routeProgress: RouteProgress) {
+        let step = routeProgress.currentLegProgress.upComingStep ?? routeProgress.currentLegProgress.currentStep
+        let nextStep = routeProgress.currentLegProgress.stepAfter(step)
+        let shouldShowNextBanner = nextStep != nil ? nextStep!.distance < 1000 : false
```
Comment (bsudekum) [inline MapboxNavigation/RouteMapViewController.swift]: Can we guard `let nextStep`?

Comment (bsudekum) [inline MapboxNavigation/RouteMapViewController.swift]: Let's use `RouteControllerHighAlertInterval` as the constant here.  https://github.com/mapbox/mapbox-navigation- ios/blob/master/MapboxCoreNavigation/Constants.swift#L95

(Maybe we should rename this 😬 )

**Comment by ericrwolfe** [comment]
@frederoni I think the shield in the then banner could be smaller, maybe 60-75% of the primary banner label's shields?

File: MapboxNavigation/RouteMapViewController.swift (hunk 1 above)
Comment (frederoni) [reply MapboxNavigation/RouteMapViewController.swift]: Can we reuse `RouteControllerMinimumDistanceForContinueInstruction` ?

Comment (bsudekum) [reply MapboxNavigation/RouteMapViewController.swift]: This is a little different. I think we will end up using this UI for continue instructions, but for a first pass to make this sync well with the `then`'s coming from the server we should only show this UI when linking instructions with `then`.

File: MapboxNavigation/RouteMapViewController.swift [hunk 2]
```diff
@@ -388,6 +388,35 @@ class RouteMapViewController: UIViewController {
         instructionsBannerView.maneuverView.step = routeProgress.currentLegProgress.upComingStep
     }
     
+    func updateNextBanner(routeProgress: RouteProgress) {
+        guard let step = routeProgress.currentLegProgress.upComingStep,
+            routeProgress.currentLegProgress.currentStepProgress.durationRemaining <= RouteControllerHighAlertInterval,
+            let nextStep = routeProgress.currentLegProgress.stepAfter(step)
```
Comment (bsudekum) [inline MapboxNavigation/RouteMapViewController.swift]: This should actually be `RouteControllerHighAlertInterval * RouteControllerLinkedInstructionBufferMultiplier `

**Comment by bsudekum** [comment]
Added then in https://github.com/mapbox/mapbox-navigation-ios/pull/819/commits/f10c72989b0d6a07ef10ea57697e653ccf10e9b4

File: MapboxNavigation/NextBannerView.swift [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    }
+    
+    func setupViews() {
+        let maneuverView = ManeuverView()
+        maneuverView.translatesAutoresizingMaskIntoConstraints = false
+        addSubview(maneuverView)
+        self.maneuverView = maneuverView
+        
+        let instructionLabel = NextInstructionLabel()
+        instructionLabel.shieldHeight = NextInstructionLabel.appearance().font.pointSize
```
Comment (bsudekum) [inline MapboxNavigation/NextBannerView.swift]: Tests are failing because font is nil here.

Comment (bsudekum) [reply MapboxNavigation/NextBannerView.swift]: @frederoni hopefully fixed with https://github.com/mapbox/mapbox-navigation-ios/pull/819/commits/d8b34885ffee4af2e86e869a28b267f8f76563af

**Pull request merged**